### ISSUE TITLE
podCIDR are not advertised after enabling loadbalancer IP announcement 

### ISSUE THREAD

**Issue by mazzy89** [cilium/cilium#18240]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

**TL;DR;**
When enabled `bgp.announce.loadbalancerIP: true` on the latest Helm chart (v1.11.0) after deployed Cilium w/ `bgp.announce.podCIDR: true`, the pod CIDR is not anymore advertised and only the loadbalancer `Service` IPs are.

 I have tested the new BGP features landed on the latest Cilium (v1.11). enabling only `podCIDR:true` on the Helm Chart the pod CIDR were advertised correctly.

Once enabled also `loadbalancerIP: true` keeping `podCIDR:true`, only the loadbalancer IPs are advertised.

I've read the doc and I'm pretty sure to understand that the two things are not mutual exclusive.

### Cilium Version

```
Client: 1.11.0 27e0848 2021-12-05T15:34:41-08:00 go version go1.17.3 linux/arm64
Daemon: 1.11.0 27e0848 2021-12-05T15:34:41-08:00 go version go1.17.3 linux/arm64
```

### Kernel Version

```
master01 [~]$ uname -a
Linux 5.10.63-v8 #1 SMP PREEMPT Tue Sep 14 08:07:05 UTC 2021 aarch64 GNU/Linux
```

### Kubernetes Version

```
❯ kubectl version
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"clean", BuildDate:"2021-10-27T18:34:20Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.5+k3s2", GitCommit:"724ef700bab896ff252a75e2be996d5f4ff1b842", GitTreeState:"clean", BuildDate:"2021-10-05T19:59:29Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/arm64"}
```

### Relevant log output

```shell
+ cilium-operator-696c5554d6-kf7nw › cilium-operator
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Started gops server" address="127.0.0.1:9891" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Auto-set \"synchronize-k8s-services\" to `true` because BGP support requires synchronizing services." subsys=option
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --bgp-announce-lb-ip='true'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --bgp-config-path='/var/lib/cilium/bgp/config.yaml'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --ces-max-ciliumendpoints-per-ces='100'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --ces-slice-mode='cesSliceModeIdentity'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --cilium-endpoint-gc-interval='5m0s'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --cluster-id=''" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --cluster-name='default'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --cluster-pool-ipv4-cidr=''" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --cluster-pool-ipv4-mask-size='24'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --cluster-pool-ipv6-cidr=''" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --cluster-pool-ipv6-mask-size='112'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --cmdref=''" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --cnp-node-status-gc-interval='2m0s'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --cnp-status-update-interval='1s'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --config=''" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --config-dir='/tmp/cilium/config-map'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --debug='false'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --disable-cnp-status-updates='true'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --disable-endpoint-crd='false'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --enable-cilium-endpoint-slice='false'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --enable-ipv4='true'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --enable-ipv4-egress-gateway='false'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --enable-ipv6='false'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --enable-k8s-api-discovery='false'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --enable-k8s-endpoint-slice='true'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --enable-k8s-event-handover='false'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --enable-local-redirect-policy='false'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --enable-metrics='false'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --gops-port='9891'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --identity-allocation-mode='crd'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --identity-gc-interval='15m0s'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --identity-gc-rate-interval='1m0s'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --identity-gc-rate-limit='2500'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --identity-heartbeat-timeout='30m0s'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --ipam='kubernetes'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --k8s-api-server=''" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --k8s-heartbeat-timeout='30s'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --k8s-kubeconfig-path=''" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --k8s-namespace='kube-system'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --k8s-service-proxy-name=''" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --kvstore=''" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --kvstore-opt='map[]'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --leader-election-lease-duration='15s'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --leader-election-renew-deadline='10s'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --leader-election-retry-period='2s'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --limit-ipam-api-burst='4'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --limit-ipam-api-qps='20'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --log-driver=''" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --log-opt='map[]'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --nodes-gc-interval='2m0s'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --operator-api-serve-addr='127.0.0.1:9234'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --operator-prometheus-serve-addr=':6942'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --parallel-alloc-workers='50'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --pprof='false'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --pprof-port='6061'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --skip-crd-creation='false'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --subnet-ids-filter=''" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --subnet-tags-filter=''" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --synchronize-k8s-nodes='true'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --synchronize-k8s-services='true'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --unmanaged-pod-watcher-interval='15'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="  --version='false'" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Cilium Operator 1.11.0 27e0848 2021-12-05T15:34:41-08:00 go version go1.17.3 linux/arm64" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Starting apiserver on address 127.0.0.1:9234" subsys=cilium-operator-api
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Establishing connection to apiserver" host="https://192.168.159.10:6443" subsys=k8s
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Connected to apiserver" subsys=k8s
cilium-operator-696c5554d6-kf7nw cilium-operator level=warning msg="Unable to discover API groups and resources" error="unable to retrieve the complete list of server APIs: metrics.k8s.io/v1beta1: the server is currently unable to handle the request" subsys=k8s
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="CRD (CustomResourceDefinition) is installed and up-to-date" name=CiliumExternalWorkload/v2 subsys=k8s
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="CRD (CustomResourceDefinition) is installed and up-to-date" name=CiliumIdentity/v2 subsys=k8s
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="CRD (CustomResourceDefinition) is installed and up-to-date" name=CiliumNode/v2 subsys=k8s
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="CRD (CustomResourceDefinition) is installed and up-to-date" name=CiliumEndpoint/v2 subsys=k8s
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="CRD (CustomResourceDefinition) is installed and up-to-date" name=CiliumNetworkPolicy/v2 subsys=k8s
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="CRD (CustomResourceDefinition) is installed and up-to-date" name=CiliumClusterwideNetworkPolicy/v2 subsys=k8s
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="attempting to acquire leader lease kube-system/cilium-operator-resource-lock..." subsys=klog
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Leader re-election complete" newLeader=worker02.hzk8s.k3s.home-fgDqFnjCxf operatorID=worker01.hzk8s.k3s.home-QARQQAOSTX subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="successfully acquired lease kube-system/cilium-operator-resource-lock" subsys=klog
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Initializing IPAM" mode=kubernetes subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Starting LB IP allocator" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Leading the operator HA deployment" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Starting to synchronize CiliumNode custom resources" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Starting to garbage collect stale CiliumEndpoint custom resources" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Starting CRD identity garbage collector" interval=15m0s subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Assigned IP \"192.0.0.1\"" event=IPAllocated
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Starting CNP derivative handler" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Starting CCNP derivative handler" subsys=cilium-operator-generic
cilium-operator-696c5554d6-kf7nw cilium-operator level=info msg="Initialization complete" subsys=cilium-operator-generic
```

### Anything else?

```
apiVersion: v1
data:
  auto-direct-node-routes: "false"
  bgp-announce-lb-ip: "true"
  bgp-announce-pod-cidr: "true"
  bpf-lb-external-clusterip: "false"
  bpf-lb-map-max: "65536"
  bpf-map-dynamic-size-ratio: "0.0025"
  bpf-policy-map-max: "16384"
  cgroup-root: /run/cilium/cgroupv2
  cilium-endpoint-gc-interval: 5m0s
  cluster-id: ""
  cluster-name: default
  custom-cni-conf: "false"
  debug: "false"
  disable-cnp-status-updates: "true"
  disable-endpoint-crd: "false"
  disable-envoy-version-check: "true"
  enable-auto-protect-node-port-range: "true"
  enable-bandwidth-manager: "false"
  enable-bpf-clock-probe: "true"
  enable-endpoint-health-checking: "true"
  enable-health-check-nodeport: "true"
  enable-health-checking: "true"
  enable-hubble: "true"
  enable-ipv4: "true"
  enable-ipv4-masquerade: "true"
  enable-ipv6: "false"
  enable-ipv6-masquerade: "true"
  enable-k8s-terminating-endpoint: "true"
  enable-l2-neigh-discovery: "true"
  enable-l7-proxy: "true"
  enable-local-redirect-policy: "false"
  enable-policy: default
  enable-remote-node-identity: "true"
  enable-session-affinity: "true"
  enable-well-known-identities: "false"
  enable-xt-socket-fallback: "true"
  hubble-disable-tls: "false"
  hubble-listen-address: :4244
  hubble-socket-path: /var/run/cilium/hubble.sock
  hubble-tls-cert-file: /var/lib/cilium/tls/hubble/server.crt
  hubble-tls-client-ca-files: /var/lib/cilium/tls/hubble/client-ca.crt
  hubble-tls-key-file: /var/lib/cilium/tls/hubble/server.key
  identity-allocation-mode: crd
  install-iptables-rules: "true"
  install-no-conntrack-iptables-rules: "false"
  ipam: kubernetes
  k8s-require-ipv4-pod-cidr: "true"
  kube-proxy-replacement: strict
  kube-proxy-replacement-healthz-bind-address: ""
  monitor-aggregation: medium
  monitor-aggregation-flags: all
  monitor-aggregation-interval: 5s
  node-port-bind-protection: "true"
  operator-api-serve-addr: 127.0.0.1:9234
  preallocate-bpf-maps: "false"
  sidecar-istio-proxy-image: cilium/istio_proxy
  tunnel: vxlan
kind: ConfigMap
metadata:
  annotations:
    meta.helm.sh/release-name: cilium
    meta.helm.sh/release-namespace: kube-system
  creationTimestamp: "2021-10-15T09:14:19Z"
  labels:
    app.kubernetes.io/managed-by: Helm
  name: cilium-config
  namespace: kube-system
  resourceVersion: "16488947"
  uid: 3591c8d2-dfd6-4cd5-b5a3-53cb9b38c739
```

```
192.0.0.1/32       via 192.168.159.10 on br41 [master01 13:01:21] * (100/0) [?]
                   via 192.168.159.14 on br41 [worker02 13:06:14] (100/0) [?]
                   via 192.168.159.13 on br41 [worker01 13:01:22] (100/0) [?]
                   via 192.168.159.15 on br41 [worker03 13:01:22] (100/0) [?]
192.168.159.0/24   dev br41 [direct1 2021-12-10] * (240)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mazzy89** [comment]
in my cilium pods however I have the log saying

```
cilium-mrp5m cilium-agent level=info msg="Advertising CIDRs to all available session" advertisements="[10.107.1.0/24]" component=MetalLBSpeaker.announcePodCidrs subsys=bgp-speaker
```

so I think routes are advertized correctly.

**Comment by mazzy89** [comment]
Despite that however when I look my bird daemon (which resides in my router and receives the routes) I get only 1 import received and consequently 1 only added. So I do not get what happen to the pod CIDR route advertized by each of the cluster workers.

```
master01 BGP      master   up     23:47:44    Established
  Description:    BGP k3s Cilium Master
  Preference:     100
  Input filter:   ACCEPT
  Output filter:  REJECT
  Routes:         1 imported, 0 exported, 1 preferred
  Route change stats:     received   rejected   filtered    ignored   accepted
    Import updates:              1          0          0          0          1
    Import withdraws:            0          0        ---          0          0
    Export updates:              3          2          1        ---          0
    Export withdraws:            0        ---        ---        ---          0
  BGP state:          Established
    Neighbor address: 192.168.159.10
    Neighbor AS:      64512
    Neighbor ID:      192.168.159.10
    Neighbor caps:    AS4
    Session:          internal multihop AS4
    Source address:   192.168.10.1
    Hold timer:       70/90
    Keepalive timer:  14/30

worker01 BGP      master   up     23:47:44    Established
  Description:    BGP k3s Cilium Worker01
  Preference:     100
  Input filter:   ACCEPT
  Output filter:  REJECT
  Routes:         1 imported, 0 exported, 0 preferred
  Route change stats:     received   rejected   filtered    ignored   accepted
    Import updates:              1          0          0          0          1
    Import withdraws:            0          0        ---          0          0
    Export updates:              3          2          1        ---          0
    Export withdraws:            0        ---        ---        ---          0
  BGP state:          Established
    Neighbor address: 192.168.159.13
    Neighbor AS:      64512
    Neighbor ID:      192.168.159.13
    Neighbor caps:    AS4
    Session:          internal multihop AS4
    Source address:   192.168.10.1
    Hold timer:       74/90
    Keepalive timer:  23/30

worker02 BGP      master   up     23:47:44    Established
  Description:    BGP k3s Cilium Worker02
  Preference:     100
  Input filter:   ACCEPT
  Output filter:  REJECT
  Routes:         1 imported, 0 exported, 0 preferred
  Route change stats:     received   rejected   filtered    ignored   accepted
    Import updates:              1          0          0          0          1
    Import withdraws:            0          0        ---          0          0
    Export updates:              4          3          1        ---          0
    Export withdraws:            0        ---        ---        ---          0
  BGP state:          Established
    Neighbor address: 192.168.159.14
    Neighbor AS:      64512
    Neighbor ID:      192.168.159.14
    Neighbor caps:    AS4
    Session:          internal multihop AS4
    Source address:   192.168.10.1
    Hold timer:       70/90
    Keepalive timer:  19/30

worker03 BGP      master   up     23:47:44    Established
  Description:    BGP k3s Cilium Worker03
  Preference:     100
  Input filter:   ACCEPT
  Output filter:  REJECT
  Routes:         1 imported, 0 exported, 0 preferred
  Route change stats:     received   rejected   filtered    ignored   accepted
    Import updates:              1          0          0          0          1
    Import withdraws:            0          0        ---          0          0
    Export updates:              3          2          1        ---          0
    Export withdraws:            0        ---        ---        ---          0
  BGP state:          Established
    Neighbor address: 192.168.159.15
    Neighbor AS:      64512
    Neighbor ID:      192.168.159.15
    Neighbor caps:    AS4
    Session:          internal multihop AS4
    Source address:   192.168.10.1
    Hold timer:       59/90
    Keepalive timer:  15/30
```

**Comment by ldelossa** [comment]
Dupe: https://github.com/cilium/cilium/issues/18250 

Thanks for details, known issue. Closing this to track in #18250

**Comment by mazzy89** [comment]
thanks @ldelossa and sorry for the dup

**Comment by ldelossa** [comment]
no worries at all @mazzy89 - the debug info is appreciated regardless.
